.github/workflows/coverage.yml

```diff
@@ -1,6 +1,6 @@
 name: Code Coverage Assessment
 # Run in addition to unit tests without coverage assessment in case of weirdness around
-# parallelised code
+# parallelised code. Also regenerates coverage badge.
 
 on:
   push:
```

Comment:
Does this want to trigger on each push or only at merge? We could do something like [this](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-pull_request_target-workflow-when-a-pull-request-merges), then use the [`checkout`](https://github.com/actions/checkout) action to add the coverage commit on top of the pull if required?

Reply:
To minimise the number of bot commits, it only commits an updated badge config when there is a change. The additional processing is only a second or so, so I'm not worried about regenerating the badge config each time. I'd also feel happier as a reviewer knowing exactly what is going to be merged.

I tried using `git push` relying on `checkout`. However, the commits aren't signed and would require "signing" the CLA on behalf of whatever email address we give to the bot. Using gh API is the way to get them signed properly.